configure.ac

```diff
@@ -1015,11 +1015,18 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <stdint.h>
 
 dnl LevelDB platform checks
 AC_MSG_CHECKING(for fdatasync)
+case $host in
+  *netbsd*)
+AC_MSG_RESULT(no because NetBSD will not fdatasync on directories and this breaks leveldb); HAVE_FDATASYNC=0
```

Comment:
Isn't there a better way to detect this condition?

Reply:
Yes; we can detect it by running fdatasync() on a directory that we know exists and observing the result.

Reply:
I considered that, but it won't work if we're cross-compiling.

BTW, *how* does it break LevelDB?

Reply:
LevelDB, if HAVE_FDATASYNC is enabled, attempts it on full directories. When bitcoind attempts to commit data to (any) leveldb it issues fdatasync on a directory, which fails and generates an incomprehensible error for the user. bitcoind then exits. Without invasive changes in the leveldb code which special-case NetBSD (or.. platforms where fdatasync on a directory won't work) then addressing this "properly" won't happen. Easier to just disable it.